utility/helpers.py

```diff
@@ -0,0 +1,78 @@
+import psycopg2
+from psycopg2.pool import ThreadedConnectionPool
+from psycopg2.extras import DictCursor
+from pathlib import Path
+from flask import g
+
+
+########### Helpers to work with one connection #####################
+
+def open_conn(host='localhost', database='phonebook', user='postgres', password=1):
+    conn = psycopg2.connect(host=host,
+                            database=database,
+                            user=user,
+                            password=1)
+
+    cur = conn.cursor(cursor_factory=DictCursor)
+    return cur, conn
+
+
+def close_conn(cur, conn):
+    conn.commit()
+    cur.close()
+    conn.close()
+
+
+##################### Connection Pooling ###############
+
+class get_cursor():
+    def __init__(self, minconn, maxconn):
+        self.bag = []
+        self.host = 'localhost'
+        self.database = 'phonebook'
+        self.user = 'postgres'
+        self.password = 1
+        self.minconn = minconn
+        self.maxconn = maxconn
+        self.connections = ThreadedConnectionPool(self.minconn,
+                                                  self.maxconn,
+                                                  host=self.host,
+                                                  database=self.database,
+                                                  user=self.user,
+                                                  password=self.password)
+
+    def __enter__(self):
+        conns = self.connections
+        conn = conns.getconn()
+        self.bag.append(conn)
+        cur = conn.cursor(cursor_factory=DictCursor)
+        return cur
+
+    def run_and_fetch(self, query_text, vals=''):
```

Comment:
Your class already implements `__enter__` and `__exit__` functions, so you shouldn't need this method to run queries. You could simply use this class with a `with` statement.

For a more in-depth explanation you can take a look at this [SO question](https://stackoverflow.com/questions/1984325/explaining-pythons-enter-and-exit).